crates/stackable-operator/src/commons/listener.rs

```diff
@@ -100,14 +123,28 @@ pub struct ListenerSpec {
     pub ports: Option<Vec<ListenerPort>>,
 
     /// Whether incoming traffic should also be directed to Pods that are not `Ready`.
-    #[schemars(default = "Self::default_publish_not_ready_addresses")]
+    #[serde(default = "ListenerSpec::default_publish_not_ready_addresses")]
     pub publish_not_ready_addresses: Option<bool>,
+
+    /// `externalTrafficPolicy` that should be set on the [`Service`] object.
+    ///
+    /// The default is `Local` (in contrast to `Cluster`), as we try pretty hard to shove traffic onto the right node,
+    /// and we should keep testing that as the primary configuration. Cluster is a fallback option for providers that
+    /// break Local mode (IONOS so far).
+    #[derivative(Default(value = "KubernetesTrafficPolicy::Local"))]
+    #[serde(default = "ListenerSpec::default_service_external_traffic_policy")]
+    pub service_external_traffic_policy: KubernetesTrafficPolicy,
 }
 
 impl ListenerSpec {
     const fn default_publish_not_ready_addresses() -> Option<bool> {
         Some(true)
     }
+
+    // `#[serde(default)]` only supports functions, so we need to wrap the literal in a function :)
+    fn default_service_external_traffic_policy() -> KubernetesTrafficPolicy {
+        Self::default().service_external_traffic_policy
```

Comment:
It's overkill to construct the whole object just to take the SETP, better to have the derivative default reference the method too instead.

Reply:
Didn't get derivative to pull a function value at the first attempt. It now works, so 0ff36cf